### DISCUSSION TITLE
How to get DJ passwords for login validation proposes to build a DJ PANEL for AzuraCast on Docker

### DISCUSSION THREAD

**Discussion by BornKillaz** [AzuraCast/AzuraCast#7074] [Q&A]
Hi,

Since there is no dedicated panel for DJs/streamers on AzuraCast, I'm forced to build one from scratch.

The main difficulty I'm facing now is accessing a DJ password for login validation, as my panel would ideally use the stream password to also access this panel, simplifying things for both me and the DJs.

I've noticed that accessing the AzuraCast MariaDB, where I believe the DJ passwords are stored, is not possible outside of Docker.

So, I was wondering if there is any other option to access their passwords. The API does not return the DJ password, and that's understandable.

The main objective of my project is to allow DJ's to upload their pre-recorded shows to be streamed later, all without allowing them to access all the station music files and folders, wich could easily be done with AzuraCast roles but also can turn into a complete disaster.

Thank you.

**Comment by BusterNeece** [comment]
Currently, DJ passwords are stored as salted hashes in the database (much the same way user passwords are stored, using a deliberately slow and secure hashing method to make brute-forcing more difficult). That means that even we don't actually know what the passwords are in the DB, and the only way we verify a new password is to hash it to the same standard and compare the hashes. You can always *set* a new password in the API, but you can't read the current password back as it's considered protected information.

It seems like several folks are building out DJ-specific portal pages for various purposes. It may be useful for us to add an API endpoint called `verify` that just verifies the password you supply it against the hashed one and returns a boolean value back. It would require an API key to access (or would otherwise have to be rate-limited much like our public login page) but it would solve the problem of verifying against AzuraCast databases when DJs are logging in elsewhere. Thoughts?

**Reply by BornKillaz** [reply]
Hi! Yes, that would be perfect!

For my project, I ended up creating an internal password system stored in MySQL. Along with SFTP and the AzuraCast API, it enabled me to build a platform exclusively for DJs. Now they can upload their pre-recorded shows without access to anything else but their own stuff and schedules.

The AzuraCast roles system just doesn't fit this purpose. But yeah, it's done, and anything else that we can get from AzuraCast regarding this type of functionality is great.

Also, it would be useful to be able to get information about the playlists that are directly related to folders and to which folder or folders. I don't think that's available yet via the API.

Thank you for all the efforts.
